File: plumbot-bukkit/src/main/java/me/regadpole/plumbot/event/server/ServerManager.java
package me.regadpole.plumbot.event.server;

import me.regadpole.plumbot.PlumBot;
import me.regadpole.plumbot.config.Config;
import me.regadpole.plumbot.tool.TextToImg;
import org.bukkit.Bukkit;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;
import java.util.concurrent.atomic.AtomicReference;

public class ServerManager {

    public static String listOnlinePlayer() {
        List<String> onlinePlayer = new LinkedList<>();
        for (Player p : Bukkit.getOnlinePlayers()) {
            onlinePlayer.add(p.getName());
        }
        return Arrays.toString(onlinePlayer.toArray()).replace("\\[|\\]", "");
    }

    public static List<String> msgList = new LinkedList<>();

    public static String sendCmd(String cmd, boolean disp) {
        AtomicReference<String> returnStr = new AtomicReference<>("无返回值");
        if(!Config.CMD()){
            returnStr.set("未开启CMD命令功能");
        }

        CommandSender commandSender = new ConsoleSender();

        PlumBot.getScheduler().runTask(() -> {
            msgList.clear();
            Bukkit.dispatchCommand(commandSender, cmd);
        });
        PlumBot.getScheduler().runTaskLaterAsynchronously(() -> {
            StringBuilder stringBuilder = new StringBuilder();
            if (msgList.size() == 0) {
                msgList.add("无返回值");
            }
            for (String msg : msgList) {
                if (msgList.get(msgList.size() - 1).equalsIgnoreCase(msg)) {
                    stringBuilder.append(msg);
                } else {
                    stringBuilder.append(msg).append("\n");
                }
            }
            if(!disp){
                msgList.clear();
                returnStr.set("无返回值");
            }
            if(stringBuilder.toString().length()<=5000){
//                PlumBot.getBot().sendMsg(isGroup, stringBuilder.toString(), id);
                switch (Config.getBotMode()) {
                    case "go-cqhttp":
                        returnStr.set(TextToImg.toImgCQCode(stringBuilder.toString()));
                        break;
                    case "kook":
                        returnStr.set(TextToImg.toImgBinary(stringBuilder.toString()));
                        break;
                    default:
                        break;
                }
            }else {
                returnStr.set("返回值过长");
            }
            msgList.clear();
        }, 10L);
        return returnStr.get();
    }

}

File: bukkit/src/main/kotlin/me/regadpole/plumbot/bukkit/util/ServerManager.kt
package me.regadpole.plumbot.bukkit.util

import me.regadpole.plumbot.PlumBot
import me.regadpole.plumbot.bot.KookBot
import me.regadpole.plumbot.bot.QQBot
import me.regadpole.plumbot.bukkit.listener.ConsoleSender
import me.regadpole.plumbot.tool.TextToImg
import org.bukkit.Bukkit
import taboolib.common.platform.function.submitAsync

object ServerManager {

    @JvmField
    var msgList: MutableList<String> = java.util.LinkedList()

    fun sendCmd(cmd: String, disp: Boolean): String {
        val returnStr = java.util.concurrent.atomic.AtomicReference("无返回值")

        val commandSender: org.bukkit.command.CommandSender = ConsoleSender()

        submitAsync(now = true) {
            msgList.clear()
            Bukkit.dispatchCommand(commandSender, cmd)
        }

        submitAsync(delay = 20L) {
            synchronized(returnStr) {
                (returnStr as Any).notify()
                val stringBuilder = java.lang.StringBuilder()
                if (msgList.size == 0) {
                    msgList.add("无返回值")
                }
                for (msg in msgList) {
                    if (msgList[msgList.size - 1].equals(msg, ignoreCase = true)) {
                        stringBuilder.append(msg)
                    } else {
                        stringBuilder.append(msg).append("\n")
                    }
                }
                if (!disp) {
                    msgList.clear()
                    returnStr.set("无返回值")
                }
                if (stringBuilder.toString().length <= 5000) {
                    //                PlumBot.getBot().sendMsg(isGroup, stringBuilder.toString(), id);
                    when (PlumBot.getBot()) {
                        is QQBot -> returnStr.set(TextToImg.toImgCQCode(stringBuilder.toString()))
                        is KookBot -> returnStr.set(stringBuilder.toString())
                        else -> {}
                    }
                } else {
                    returnStr.set("返回值过长")
                }
                msgList.clear()
            }
        }

        synchronized(returnStr) {
            try {
                (returnStr as Any).wait()
            } catch (e: InterruptedException) {
                throw java.lang.RuntimeException(e)
            }
            return returnStr.get()
        }
    }
}
